sources/ghostdog/src/main.rs

```diff
@@ -63,6 +141,44 @@ fn run() -> Result<()> {
             let device_name = find_device_name(format!("{}", path.display()))?;
             emit_device_name(&device_name);
         }
+        SubCommand::MatchNvidiaDriver(vendor) => {
+            // Currently, we only want to run this for NVIDIA devices
+            if vendor.vendor == NVIDIA_VENDOR_ID {
+                let pci_id = env::var("PCI_ID").context(error::MissingPciIdEnvSnafu)?;
+                let driver_choice = find_preferred_driver(pci_id)?;
+                // After choosing the driver, drop a marker file provide this decision to others on the system (like systemd)
+                if !Path::new(NVIDIA_RUN_DIR).exists() {
```

Comment:
Should we create the directory in systemd.tmpfiles? I have seen very similar directory creation fail during boot (possibly this can race), and tmpfiles is the canonical way to get the directory structure we want.

Reply:
> Should we create the directory in systemd.tmpfiles?

tmpfiles runs rather late - after `local-fs.target` is reached - and `ghostdog` can be invoked before that.

Reply:
This is what I found in my testing, this executes so early in boot that you need to be ready to create these things yourself. They are called out in a tmpfile configuration but ghostdog beat it consistently. The same goes for when copying the modules, the invocation typically happens before so making the directory just in case ends up ensuring it is there.